src/main/resources/hudson/plugins/favorite/column/FavoriteColumn/column.jelly

```diff
@@ -1,27 +1,16 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout"
-         xmlns:t="/lib/hudson" xmlns:s="/lib/form">
-  <j:if test="${it.isLoggedIn()}">
-    <j:set var="sortData" value="${it.getSortData(job.getFullName())}"/>
-    <j:set var="starClass" value="${it.getStarClassName(job.getFullName())}"/>
-    <j:set var="sizeClass" value="${it.getSizeClassFromIconSize(subIconSize)}"/>
-    <td data="${sortData}">
-      <a href="#">
-        <span id="fav_${job.getFullName()}" data-fullName="${job.getFullName()}"
-              class="${starClass} ${sizeClass}"/>
+<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout">
+  <j:set var="sortData" value="${it.getSortData(job.getFullName())}"/>
+  <j:set var="fav" value="${it.isFavorite(job.getFullName())}"/>
+  <j:set var="sizeClass" value="${icons.toNormalizedIconSizeClass(subIconSize)}"/>
+  <td data="${sortData}" class="jenkins-table__cell--tight">
+    <div class="jenkins-table__cell__button-wrapper">
+      <a href="#" class="jenkins-table__button" onclick="toggleFavoriteOnClick(event, this)" data-fullName="${job.getFullName()}">
+        <span id="fav_${job.getFullName()}" data-fav="${fav}">
+            <l:icon src="symbol-star plugin-ionicons-api" class="icon-fav-active jenkins-!-color-yellow ${sizeClass} ${!fav ? 'jenkins-hidden' : ''}"/>
```

Comment:
Will that potentially cause some flickering as you hide on icon and show another?
Maybe always use the `symbol-star` and just change the color in js by adding removing the class `jenkins-!-color-yellow`

Reply:
There is no noticeable flickering that I've observed. I originally has the solid star and changed the color but after testing it was unclear that the non-yellow star meant that it was not toggled. The empty star makes the state more clear in my opinion.